### PR TITLE
SIMSBIOHUB-424: Publish Documents to BioHub from SIMS

### PR DESCRIPTION
**Pull request by KjartanE**
## Links to Jira Tickets
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-424

## Description of Changes
- Send artifacts to BioHub as part of the existing publish workflow.
- The artifacts are to be sent as separate requests 
- When a survey has attached artifacts, those artifacts are submitted to biohub during publish

## Testing Notes

- {List any relevant testing considerations, necessary pre-reqs, and areas of the app to focus on. Specifically, include anything that will help the reviewers of this PR verify the code is functioning as expected.}

Related PR for the BioHub changes: https://github.com/bcgov/biohubbc-platform/pull/230

### CODE REVIEW

File: api/src/paths/project/{projectId}/survey/{surveyId}/view.ts [hunk 1]
```diff
@@ -399,9 +399,8 @@ GET.apiDoc = {
                         oneOf: [{ type: 'object' }, { type: 'string', format: 'date' }],
                         description: 'ISO 8601 date string for the project start date'
                       },
-                      queue_id: {
-                        type: 'integer',
-                        minimum: 1
+                      submission_uuid: {
+                        type: 'string'
```
Comment (curtisupshall) [inline api/src/paths/project/{projectId}/survey/{surveyId}/view.ts]: Nit: add `format: 'uuid'`

File: api/src/paths/project/{projectId}/view.ts [hunk 2]
```diff
@@ -213,9 +213,8 @@ GET.apiDoc = {
                         oneOf: [{ type: 'object' }, { type: 'string', format: 'date' }],
                         description: 'ISO 8601 date string for the project start date'
                       },
-                      queue_id: {
-                        type: 'integer',
-                        minimum: 1
+                      submission_uuid: {
+                        type: 'string'
```
Comment (curtisupshall) [inline api/src/paths/project/{projectId}/view.ts]: Nit: add `format: 'uuid'`

File: api/src/paths/publish/survey.ts [hunk 3]
```diff
@@ -71,8 +71,9 @@ POST.apiDoc = {
           schema: {
             type: 'object',
             properties: {
-              submission_id: {
-                type: 'number'
+              submission_uuid: {
+                type: 'string',
+                description: 'The UUID of the submission'
```
Comment (curtisupshall) [inline api/src/paths/publish/survey.ts:78]: Nit: add `format: 'uuid'`

File: api/src/repositories/history-publish-repository.ts [hunk 4]
```diff
@@ -6,49 +6,49 @@ import { BaseRepository } from './base-repository';
...
-  artifact_id: number;
+  artifact_uuid: string;
 }
 
 export const ProjectMetadataPublish = z.object({
   project_metadata_publish_id: z.number(),
   project_id: z.number(),
   event_timestamp: z.string(),
-  queue_id: z.number(),
+  submission_uuid: z.string(),
```
Comment (curtisupshall) [inline api/src/repositories/history-publish-repository.ts]: Can use `z.string().uuid()`

**Review by curtisupshall — DISMISSED** [review]
LGTM

**Review by curtisupshall — DISMISSED** [review]
Static read of the code looks good to me 📑

**Review by curtisupshall — APPROVED** [review]
LGTM

**Review by al-rosenthal — APPROVED** [review]
ran this locally and it works as expected! 🌮

**Pull request merged**